### DISCUSSION TITLE
Long file names

### DISCUSSION THREAD

**Discussion by AlexZIX** [minio/minio#18571] [Q&A] [answered]
Hi,

few days ago we got stuck with limit for file name length. S3 allows the file name length up to 1024 characters but xfs which recommended for minio setups and other filesystems allows only 255 ASCII characters. So while we trying to upload file with Russian name and 180 characters length of its UTF-8 name we got error that file name is too long. 
I've googled information about the limits in other filesystems and all systems have the same limit of 255 characters. So how we can upload files to minio with more than 255 characters length. For now I see the only way to patching linux kernel. Or may be more easier way exists?

**Comment by harshavardhana** [comment, marked as answer]
Unfortunately, that is not possible, save the actual key in a DB and save a UUID reference on MinIO - it is much more ideal that way.
